homeassistant/components/screenlogic/config_flow.py

```diff
@@ -58,14 +58,11 @@ def name_for_mac(mac):
 
 async def async_get_mac_address(hass, ip_address, port):
     """Connect to a screenlogic gateway and return the mac address."""
-    connected_socket = await hass.async_add_executor_job(
-        login.create_socket,
-        ip_address,
-        port,
-    )
-    if not connected_socket:
-        raise ScreenLogicError("Unknown socket error")
-    return await hass.async_add_executor_job(login.gateway_connect, connected_socket)
+    transport, protocol = await login.async_create_connection(ip_address, port)
+    mac = await login.async_gateway_connect(transport, protocol)
+    if transport and not transport.is_closing():
+        transport.close()
+    return mac
```

Comment:
Can you move this logic into the library itself so we don't have to test it directly in Home Assistant?

Reply:
That's probably best. Having a hard time getting it to hit Ln64.

Reply:
https://github.com/home-assistant/core/pull/60466/commits/063753153a87caa9e4680db53d42b81377114603